.kokoro/build.sh

```diff
@@ -0,0 +1,237 @@
+    cbt -project="${BIGTABLE_PROJECT_ID}" -instance="${BIGTABLE_INSTANCE_ID}" \
+            createtable "${TABLE_ID}" "families=col_family" splits="$split_points"
+}
+
+delete_table() {
+    BIGTABLE_PROJECT_ID=$1
+    BIGTABLE_INSTANCE_ID=$2
+    TABLE_ID=$3
+    cbt -project=${BIGTABLE_PROJECT_ID} -instance=${BIGTABLE_INSTANCE_ID} \
+        deletetable  ${TABLE_ID}
+    if [[ $? != 0 ]]; then echo "Could not delete table for PySpark test."; fi
+}
+
+run_load_test() {
+    echo "***Running Load test.***"
+    BIGTABLE_SPARK_JAR=$(get_bigtable_spark_jar)
+    TEST_SCRIPT="spark-bigtable_2.12/test-pyspark/load_test.py"
+    BASE_SCRIPT="spark-bigtable_2.12/test-pyspark/test_base.py"
+    TABLE_ID=$(create_table_id "load")
+    gcloud dataproc jobs submit pyspark \
+        --cluster=${DATAPROC_CLUSTER_NAME} \
+        --region=${DATAPROC_CLUSTER_REGION} \
+        --jars=${BIGTABLE_SPARK_JAR} \
+        ${TEST_SCRIPT} \
+        --py-files=${BASE_SCRIPT} \
+        -- \
+        --bigtableProjectId=${BIGTABLE_PROJECT_ID} \
+        --bigtableInstanceId=${BIGTABLE_INSTANCE_ID} \
+        --bigtableTableId=${TABLE_ID}
+    exit_code=$?
+    delete_table ${BIGTABLE_PROJECT_ID} ${BIGTABLE_INSTANCE_ID} ${TABLE_ID}
+    return $exit_code
+}
+
+run_fuzz_tests() {
+    SPARK_VERSION=$1
+    echo "***Running Spark-Bigtable fuzz tests for Spark ${SPARK_VERSION}.***"
+    TABLE_ID=$(create_table_id "fuzz")
+    create_table_with_random_splits \
+      "${BIGTABLE_PROJECT_ID}" "${BIGTABLE_INSTANCE_ID}" "$TABLE_ID"
+    BIGTABLE_SPARK_IT_MODULE="spark-bigtable_2.12-it"
+    mvn -pl ${BIGTABLE_SPARK_IT_MODULE} \
+        failsafe:integration-test failsafe:verify \
+        -B -ntp -Dclirr.skip=true -Denforcer.skip=true -Dcheckstyle.skip \
+        -Dspark.version="${SPARK_VERSION}" \
+        -DbigtableProjectId="${BIGTABLE_PROJECT_ID}" \
+        -DbigtableInstanceId="${BIGTABLE_INSTANCE_ID}" \
+        -DbigtableTableId="${TABLE_ID}" \
+        -P fuzz
+    exit_code=$?
+    delete_table "${BIGTABLE_PROJECT_ID}" "${BIGTABLE_INSTANCE_ID}" "${TABLE_ID}"
+    return $exit_code
+}
+
+run_pyspark_test() {
+    SPARK_VERSION=$1
+    HADOOP_VERSION=$2
+    # TODO: Install Python 3.7 on Kokoro machines to enable PySpark 2.4.8 tests
+    #   without running into the "TypeError: an integer is required" issue.
+    if [[ ${SPARK_VERSION} == "2.4.8" ]]; then return; fi
+
+    echo "***Running the PySpark test for Spark ${SPARK_VERSION}.***"
+    SPARK_BIN_NAME="spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}"
+    BIGTABLE_SPARK_JAR=$(get_bigtable_spark_jar)
+    PYSPARK_TEST_SCRIPT="spark-bigtable_2.12/test-pyspark/read_and_write_test.py"
+    BASE_SCRIPT="spark-bigtable_2.12/test-pyspark/test_base.py"
+    # Download Apache Spark on the machine from a GCS bucket, new versions
+    # must be manually uploaded to the bucket first.
+    gsutil -q cp "gs://bigtable-spark-test-resources/spark-files/${SPARK_BIN_NAME}.tgz" .
```

Comment:
> Better yet, the env setup should be automatic...if the jar is not there then upload it as a class test rule

Do you mean moving the logic for copying the jar to the test code instead of the script?

Reply:
You expect a gcs bucket to be manually prepared with the spark tarball. This makes it difficult to re-create the test environment because someone would have to hunt around for all of the expected bits. Throughout the codebase. The initial suggestion was to create a env setup script that uploads the spark distribution and creates whatever other resources you need.

The 2nd suggestion is to do it in java